### PR TITLE
CMake: implement ability to export build tree

### PR DESCRIPTION
**Pull request by mic84**

### CODE REVIEW

**Comment by mic84** [comment]
@sayerhs try this out and let me know if it works for you.

**Review by sayerhs — APPROVED** [review]

**Review by ax3l — APPROVED** [review]
Thank you both, great work! :sparkles:

File: Tools/CMake/AMReXInstallHelpers.cmake [hunk 1]
```diff
@@ -80,20 +80,30 @@ function (install_amrex)
    # Write and install configure file
    include(CMakePackageConfigHelpers)
 
+   #
+   # Setup for install and export of build tree
+   #
+   set(CMAKE_FILES_DIR   lib/cmake/AMReX)   # Relative path to top level installation/build-tree
```
Comment (ax3l) [inline Tools/CMake/AMReXInstallHelpers.cmake]: for installation, that path is only valid on Unix:
https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure
```suggestion
   # Relative path to top level installation/build-tree
   if(WIN32)
       set(CMAKE_FILES_DIR   cmake)   
   else()
       set(CMAKE_FILES_DIR   lib/cmake/AMReX)
   endif()
```

Comment (ax3l) [reply Tools/CMake/AMReXInstallHelpers.cmake]: I think one usually sets them via changed defaults in a cached `CMAKE_INSTALL_CMAKEDIR` variable:
https://github.com/openPMD/openPMD-api/blob/ddc673bcd96d28293a11cfdbbc8fdcd57c154385/CMakeLists.txt#L38

**Review by ax3l — APPROVED** [review]

**Pull request merged**